tree/ntuple/v7/src/RPageSinkBuf.cxx

```diff
@@ -31,9 +31,11 @@ ROOT::Experimental::Detail::RPageSinkBuf::RPageSinkBuf(std::unique_ptr<RPageSink
    fMetrics.ObserveMetrics(fInnerSink->GetMetrics());
 }
 
-void ROOT::Experimental::Detail::RPageSinkBuf::CreateImpl(const RNTupleModel &model)
+void ROOT::Experimental::Detail::RPageSinkBuf::CreateImpl(const RNTupleModel &model,
+                                                          unsigned char * /* serializedHeader */,
```

Comment:
Shouldn't we just avoid the overhead of serializing a header in the base `RPageSink::Create()` in this case (same for the page list and footer below)?

Reply:
True, there is the duplicated work.  Before, however, we had the duplicated code of serializing the header in each of the concrete backends.  I think overall it is an improvement but not yet ideal.  The problem is, I think, that the interplay of derived and based class is not optimally designed for page storage wrapper classes. I'll add a TODO.

Reply:
What about having (protected) functions that can be reused from derived classes to accomplish common tasks, e.g. `... RPageSink::SerializeHeader(...)`, etc.?
For instance, instead of unconditionally serializing from the base class' `RPageSink::Create()`, each concrete backend could call to `SerializeHeader()` provided by the base class, if required.  That's one non-virtual call in a non-hot-path, which should be fine.  :-)